app/scripts/controllers/network/network.js

```diff
@@ -235,6 +251,45 @@ export default class NetworkController extends EventEmitter {
   // Private
   //
 
+  async _checkInfuraAvailability(network) {
+    const rpcUrl = `https://${network}.infura.io/v3/${this._infuraProjectId}`;
+
+    let networkChanged = false;
+    this.on(NETWORK_EVENTS.NETWORK_DID_CHANGE, () => {
```

Comment:
Nit: This should probably use `once` instead of `on`, since we only care about the first time we see this

```suggestion
    this.once(NETWORK_EVENTS.NETWORK_DID_CHANGE, () => {
```

Reply:
Makes sense, fixed